src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -41,69 +61,416 @@ internal class CodeLine
+            }
+            // else there is no comment
+            else
+            {
+                this.WithoutEnding = text.Substring(0, lineLength);
+                this.EndOfLineComment = null;
+            }
+
+            this.StringDelimiters = delimiterBuilder.ToImmutable();
+            this.ErrorDelimiterPositions = errorDelimiterBuilder.ToImmutable();
+            this.Indentation = indentation;
+            this.ExcessBracketPositions = excessBrackets.ToImmutableArray();
+            ScopeTracking.VerifyStringDelimiters(text, this.StringDelimiters);
+            ScopeTracking.VerifyExcessBracketPositions(this, excessBrackets);
         }
 
-        public static CodeLine Empty() => new CodeLine(string.Empty, Enumerable.Empty<int>(), 0);
+        /// <summary>
+        /// Returns true if the given string context is indicative of being
+        /// inside any kind of string, false otherwise.
+        /// </summary>
+        private static bool IsInAString(StringContext context) => context >= StringContext.OpenString;
 
-        internal CodeLine SetText(string newText)
+        private static int FindNextDelimOrComment(string text, char[] delimiters, int startIndex)
         {
-            return new CodeLine(newText, this.StringDelimiters, this.WithoutEnding.Length, this.Indentation, this.ExcessBracketPositions);
+            var delimitersAndComment = delimiters.Append('/').ToArray();
+
+            var index = text.IndexOfAny(delimitersAndComment, startIndex);
+            if (index >= 0 && text[index] == '/')
+            {
+                // Found a comment
+                if (index + 1 < text.Length && text[index + 1] == '/')
+                {
+                    return index;
+                }
+                else
+                {
+                    // Just a lone /, move past it and try again
+                    return FindNextDelimOrComment(text, delimiters, index + 1);
+                }
+            }
+            else
+            {
+                return index;
+            }
         }
 
-        internal CodeLine SetStringDelimiters(IEnumerable<int> newStringDelimiters)
+        /// <summary>
+        /// Computes the location of the string delimiters within a given text.
+        /// </summary>
+        private static IEnumerable<int> ComputeStringDelimiters(string text, ref StringContext stringContext, out int commentIndex, out IEnumerable<int> errorDelimiters)
         {
-            return new CodeLine(this.Text, newStringDelimiters, this.WithoutEnding.Length, this.Indentation, this.ExcessBracketPositions);
+            var delimiterBuilder = ImmutableArray.CreateBuilder<int>();
+            var errorDelimiterBuilder = ImmutableArray.CreateBuilder<int>();
+
+            if (IsInAString(stringContext))
+            {
+                delimiterBuilder.Add(-1);
+            }
+
+            commentIndex = -1;
+            var noOpenStringDelimiters = new[] { '"', '/' };
```

Comment:
These are still allocated on every invocation of ComputeStringDelimiters. I suggest moving them out of the function and making them static readonly. I.e.
`        static readonly char[] noOpenStringDelimiters = new[] { '"', '/' };`

Reply:
Oh you are right, I've now made them static fields, as you suggest.